s3prl/downstream/speech_translation/dataset.py

```diff
@@ -0,0 +1,23 @@
+from torch.utils.data import Dataset
+from multiprocessing import Pool
+
+class DummyDataset(Dataset):
```

Comment:
I'm OK with the current dummy dataset settings, but would suggest a more elegant way :-)

Reply:
The functionality of this DummyDataset class to help bridge the gap between Fairseq dataset and Torch dataset used in s3prl to support batch sampling with fixed tokens per batch, which could make the training more efficient.

There is another more elegant way which is modifying 
https://github.com/s3prl/s3prl/blob/205cdc3abd7d56169a98557bf865230a661a06de/s3prl/downstream/runner.py#L184-L193
By refreshing dataloader at the beginning of each epoch, we could directly using Fairseq dataset and remove DummyDataset.
Possible modification could be done as following codes
```python
# prepare data
dataloader = self.downstream.model.get_dataloader('train') 

batch_ids = []
backward_steps = 0
records = defaultdict(list)
epoch = self.init_ckpt.get('Epoch', 0)
while pbar.n < pbar.total:
   if is_initialized():
      # dataloader.sampler.set_epoch(epoch) 
      dataloader = self.downstream.refresh_dataloader(epoch, 'train') # <- add codes here
```
if @leo19941227 are interested in this modification, we could discuss more about it.

Reply:
Hey @bearhsiang, yes! Actually I am planning to update the runner to let it re-get the dataloader from the downstream expert after each epoch. Also the `get_dataloader` function will have a new argument: `epoch: int`. Do you think this proposal can help you to make this part of code more elegant?

Reply:
Just for a comparison. Currently our pipeline get the dataloader at the start for the entire training, and then iterate over the dataloader again and again, where the dataloader and the dataset inside it can't adjust according to the epoch.

Reply:
@leo19941227 

Yes, that would be great, and we could avoid using DummyDataset in this case.

Reply:
The get_dataloader with the epoch is supported at #203